src/main/java/frc/robot/controls/GameControllerDriverConfig.java

```diff
@@ -52,42 +49,43 @@ public void configureControls() {
     controller.get(Button.START).onFalse(new BangBangSubsystemCommand(sub, 0));
 
     // TODO 4.2.2: Make robot spin while a button is pressed
-    controller.get(Button.A).whileTrue(new RunCommand(() -> drive.arcadeDrive(0, 1.0), drive));
-    controller.get(Button.Y).whileTrue(new RunCommand(() -> drive.arcadeDrive(0, -1.0), drive));
+    controller.get(Button.A).whileTrue(new RunCommand(() -> getDrivetrain().arcadeDrive(0, 1.0), getDrivetrain()));
+    controller.get(Button.Y).whileTrue(new RunCommand(() -> getDrivetrain().arcadeDrive(0, -1.0), getDrivetrain()));
 
     // TODO 4.3.1: Add more triggers
 
     // SequentialCommandGroup: Drive forward, then turn 45 degrees
     controller.get(Button.RIGHT_JOY).onTrue(new SequentialCommandGroup(
-      new RunCommand(() -> drive.arcadeDrive(0, 0), drive),    // stop
-      new AutoForwardCommand(drive, 1.0, 3.0)                  // drive forward
+     new AutoForwardCommand(getDrivetrain(), 1.0, 3.0), // drive forward
+     new AutoTurnCommand(getDrivetrain(), 63), // turn 45 degrees**
+     new InstantCommand(() -> getDrivetrain().arcadeDrive(0, 0), getDrivetrain())  // stop at the end
   ));
     // ParallelCommandGroup: Drive forward and rotate simultaneously
     controller.get(Button.LEFT_JOY).onTrue(new ParallelCommandGroup(
-        new AutoForwardCommand(drive, 1.0, 2.0),
+        new AutoForwardCommand(getDrivetrain(), 1.0, 2.0),
         new BangBangRotateCommand(sub, 1.0)
     ));
 
     // ConditionalCommand: Drive forward if LEFT_TRIGGER_BUTTON is pressed, otherwise turn
-new Trigger(controller.LEFT_TRIGGER_BUTTON).onTrue(
+  controller.get(controller.LEFT_TRIGGER_BUTTON).onTrue(
     new ConditionalCommand(
-        new AutoForwardCommand(drive, 1.0, 2.0),
-        new AutoTurnCommand(drive, -90),
-        () -> controller.LEFT_TRIGGER_BUTTON.getAsBoolean()
+      new AutoForwardCommand(getDrivetrain(), 1.0, 2.0),
+      new AutoTurnCommand(getDrivetrain(), -90),
+      controller.get(controller.LEFT_TRIGGER_BUTTON)
     )
```

Comment:
It still uses LEFT_TRIGGER_BUTTON. The condition cannot be the same as the trigger, since this would make it always true. You also don't need controller.get() here, because controller.LEFT_TRIGGER_BUTTON (or controller.RIGHT_TRIGGER_BUTTON) is already the right type. controller.get() converts it to a Trigger, and this command takes a BooleanSupplier.

Reply:
just updated the conditional command

Reply:
Now you aren't scheduling anything on the button press. The code structure should be:
controller.get(controller.LEFT_TRIGGER_BUTTON).onTrue(
new ConditionalCommand(
command1,
command2,
condition
));
condition should be a BooleanSupplier, such as controller.RIGHT_TRIGGER_BUTTON or, if something can't efficiently be converted to a BooleanSupplier, a Trigger like controller.get(DPad.something). You could also use something other than a button as a condition, like ()->getDrivetrain().getPose().getX()>5. If something is already a BooleanSupplier, like LEFT_TRIGGER_BUTTON, you don't need to use controller.get() to convert it to a Trigger.

Reply:
You still need the controller.get(controller.LEFT_TRIGGER_BUTTON).onTrue before the command so it is scheduled when it becomes true.

Reply:
Just added the scheduler. I also modified the condition you mentioned, as it lets allows the condition to actually have two cases. It's starting to make more sense now—thanks for the clarification!